src/Apps/MyCollection/Routes/UploadFlow/ArtworkDetails/MyCollectionArtworkDetails.tsx

```diff
@@ -0,0 +1,196 @@
+import { Button, Spacer, useToasts } from "@artsy/palette"
+import { UploadStepper } from "Apps/MyCollection/Components/UploadStepper"
+import { BackLink } from "Components/Links/BackLink"
+import { Form, Formik } from "formik"
+import { createFragmentContainer, graphql } from "react-relay"
+import { useSystemContext } from "System"
+import { useRouter } from "System/Router/useRouter"
+import createLogger from "Utils/logger"
+import { ArtworkAttributionClassType } from "__generated__/CreateArtworkMutation.graphql"
+import { MyCollectionArtworkDetails_artwork } from "__generated__/MyCollectionArtworkDetails_artwork.graphql"
+import { createOrUpdateArtwork } from "../Utils/createOrUpdateArtwork"
+import {
+  MyCollectionArtworkDetailsValidationSchema,
+  validate,
+} from "../Utils/validation"
+import {
+  getMyCollectionArtworkDetailsFormInitialValues,
+  MyCollectionArtworkDetailsForm,
+  MyCollectionArtworkDetailsFormModel,
+} from "./Components/MyCollectionArtworkDetailsForm"
+
+const logger = createLogger("UploadFlow/MyCollectionArtworkDetails.tsx")
+
+export interface MyCollectionArtworkDetailsProps {
+  artwork?: MyCollectionArtworkDetails_artwork
+}
+
+export const MyCollectionArtworkDetails: React.FC<MyCollectionArtworkDetailsProps> = ({
+  artwork,
+}) => {
+  const { router } = useRouter()
+  const { relayEnvironment } = useSystemContext()
+  const { sendToast } = useToasts()
+  const initialValue = getMyCollectionArtworkDetailsFormInitialValues(artwork)
+  const initialErrors = validate(
+    initialValue,
+    MyCollectionArtworkDetailsValidationSchema
+  )
+
+  const handleSubmit = async (values: MyCollectionArtworkDetailsFormModel) => {
+    const isLimitedEditionRarity = values.attributionClass === "limited edition"
+
+    const MyCollectionArtworkDetailsForm = {
+      ...values,
+      editionNumber: isLimitedEditionRarity ? values.editionNumber : "",
+      editionSize: isLimitedEditionRarity ? values.editionSize : "",
+    }
+
+    for (let key in MyCollectionArtworkDetailsForm) {
+      if (typeof MyCollectionArtworkDetailsForm[key] !== "string") {
+        MyCollectionArtworkDetailsForm[key] =
+          MyCollectionArtworkDetailsForm[key] || undefined
+      } else {
+        MyCollectionArtworkDetailsForm[key] =
+          MyCollectionArtworkDetailsForm[key] &&
+          MyCollectionArtworkDetailsForm[key].trim()
+      }
+    }
+
+    let artworkId: string
+
+    if (relayEnvironment) {
```

Comment:
There shouldn't ever be a situation in Force where `relayEnvironment` is undefined; if there is, that's a bug!

Reply:
Thanks 🙇